File: backend/src/main/java/com/classstudies/classquiz/service/UserService.java
package com.classstudies.classquiz.service;

import java.util.Set;

import com.classstudies.classquiz.model.User;
import com.classstudies.classquiz.model.UserRole;

public interface UserService {
	
	//create user
	public User createUser(User user, Set<UserRole> userRoles) throws Exception;
	
	//get user by username
	public User getUser(String username);
	
	//delete user by id
	public void deleteUser(Long userId);
	
	//get all users
	public Set<User> getUsers();
	
	// update user
	public User updateUser(Long id, User user) throws Exception;
}

File: frontend/src/app/services/user.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class UserService {

  constructor(private http: HttpClient) { } 

  //add user
  public addUser(user:any){
    return this.http.post(`${baseUrl}/user/`, user);
  }

  // get user
  public getUser(username:any){
    return this.http.get(`${baseUrl}/user/${username}`);
  }

  // get all users
  public getUsers(){
    return this.http.get(`${baseUrl}/user/`);
  }

  // delete user 
  public deleteUser(userId:any){
    return this.http.delete(`${baseUrl}/user/${userId}`);
  }

  // otp
  public verifyotp(otpData: any){
    return this.http.post(`${baseUrl}/user/v/`, otpData);
  }
}
